MediaGallerySynchronization/Model/AssetImagesSynchronization.php

```diff
@@ -5,21 +5,21 @@
  */
 declare(strict_types=1);
 
-namespace Magento\MediaGalleryUi\Model;
+namespace Magento\MediaGallerySynchronization\Model;
 
 use Magento\Framework\App\Filesystem\DirectoryList;
 use Magento\Framework\Filesystem;
 use Magento\Framework\Filesystem\Directory\ReadInterface;
 use Magento\Framework\Filesystem\Driver\File;
 use Magento\MediaGalleryApi\Api\Data\AssetInterfaceFactory;
 use Magento\MediaGalleryApi\Model\Asset\Command\SaveInterface;
-use Magento\MediaGalleryUi\Model\Filesystem\IndexerInterface;
+use Magento\MediaGallerySynchronizationApi\Api\AssetImagesSynchronizationInterface;
 use Psr\Log\LoggerInterface;
 
 /**
  * Create MediaGallery asset and save it to database based on file information
  */
-class AssetIndexer implements IndexerInterface
+class AssetImagesSynchronization implements AssetImagesSynchronizationInterface
```

Comment:
Let's simplify the name of the class and interface
```suggestion
class ImagesSynchronization implements ImagesSynchronizationInterface
```